app/routes/index.js

```diff
@@ -1,9 +1,39 @@
+require('dotenv').config();
 const express = require('express');
 const router = express.Router();
+const dotenv = require("dotenv");
+const jwt = require('jsonwebtoken');
+
+dotenv.config();
+
+process.env.TOKEN_SECRET;
 
 /* GET home page. */
 router.get('/', function(req, res, next) {
-    res.render('index', {title:'OG-YOGA'});
+    res.render('index', {title:'OG-YOGA', error: ""});
+});
+
+router.post('/login', async function(req, res, next) {
+    if(req.body.secret_key !== process.env.TOKEN_SECRET){
+        const error = "wrong secret key";
+        const token = await generateToken({username: req.body.username, error: error});
+        res.cookie('token', token, { httpOnly: true });
+        res.json({ token });
+    }else{
+        const token = await generateToken({username: req.body.username, error: ""});
+        res.cookie('token', token, { httpOnly: true });
+        res.json({ token });
+    }
 });
 
+//idk where to put this rn so I'll hold it here
+function generateToken(username, error) {
```

Comment:
Make a new folder under app called auth, and make a new js file called something like tokenUtil and move all the token functions there